src/include/gpopt/gpdbwrappers.h

```diff
@@ -303,6 +303,9 @@ namespace gpdb {
 	// find the oid of the root partition given partition oid belongs to
 	Oid OidRootPartition(Oid oid);
 	
+	// find the number of partitions for a given root relation
+	int INumberPartitions(Oid oid);
+
```

Comment:
Looked at this function again, `UlLeafPartitions` returns the number of leaf partitions, it only accept oid of root partitioned table. 
Both function uses `countLeafPartTables` to count the number of leaf partitions. 
The only difference is that `INumberPartitions ` can accept non root partition oid, just return 0 when it is neither a root partition table nor a partition table.
In fact, we can get rid of INumberPartitions.